packages/cli/src/cmds/agentInstaller/projectConfiguration.ts

```diff
@@ -81,12 +84,42 @@ async function resolveSelectedInstallers(
     }
   }
 
-  // Sanity check. Verify everything resolved properly. This should branch should never occur.
+  // Sanity check. Verify everything resolved properly. This branch should never occur.
   if (projects.some(({ selectedInstaller }) => !selectedInstaller)) {
     throw new Error('Invalid selection');
   }
 }
 
+export function getYarnPackages(availableInstallers: AgentInstaller[]): string[] {
+  let yarnPackages: string[] = [];
+  availableInstallers.forEach((installer) => {
+    if (installer.name === 'yarn') {
+      const packageJsonFilename = join(installer.path, 'package.json');
+      if (existsSync(packageJsonFilename)) {
+        const data = readFileSync(packageJsonFilename, 'utf-8');
+        const packageJson = JSON.parse(data);
+        if ('workspaces' in packageJson) {
+          const workspaces = packageJson['workspaces'];
+          // process as glob because it could be something like
+          // workspaces: [ 'packages/*' ]
+          workspaces.forEach((workspace) => {
+            const matches = glob.sync(workspace, {
+              matchBase: true,
+              cwd: installer.path,
+              ignore: ['**/node_modules/**', '**/.git/**'],
+            });
```

Comment:
Set `strict` to false here to avoid permission errors thrown by `glob`. I'd normally say this is a case we don't need to worry about, but as we've learned, edge cases are not really edge cases once this is running out in the wild.

Reply:
Added.